### ISSUE TITLE
Samsung SM-T575 / Galaxy Tab Active 3 fails to build working AP image // vbmeta hash mismatch

### ISSUE THREAD

**Issue by msmuenchen** [topjohnwu/Magisk#3746]
Magisk Manager in 21.4 and Canary (f5593e05) fail to create an AP image that can be flashed using ODIN for the (new) SM-T575 tablet. The ODIN flash fails after the device tries to boot, with the bootloader going into download mode and displaying:

    ODIN MODE (AVB Fail)
    vbmeta: Error verifying vbmeta image: HASH_MISMATCH (3)
    vbmeta: VERIFICATION_DISABLED bit is set
    CUSTOM VBMETA
    VBMETA T575XXU1ATK1, 35608539R

* Firmware: Android 10 // T575XXU1ATK1/T575OXM1ATK1/T575XXU1ATK1/T575XXU1ATK1
* Ramdisk yes, A/B no, SAR yes, Crypto File

ODIN log:

```
<ID:0/005> Added!!
<ID:0/005> Odin engine v(ID:3.1401)..
<ID:0/005> File analysis..
<ID:0/005> Total Binary size: 7178 M
<ID:0/005> SetupConnection..
<ID:0/005> Initialzation..
<ID:0/005> Set PIT file..
<ID:0/005> DO NOT TURN OFF TARGET!!
<ID:0/005> Get PIT for mapping..
<ID:0/005> Firmware update start..
<ID:0/005> NAND Write Start!! 
<ID:0/005> SingleDownload.
<ID:0/005> sboot.bin
<ID:0/005> up_param.bin
<ID:0/005> cm.bin
<ID:0/005> keystorage.bin
<ID:0/005> vbmeta.img
<ID:0/005> recovery.img
<ID:0/005> dtbo.img
<ID:0/005> super.img
<ID:0/005> userdata.img
<ID:0/005> vbmeta.img
<ID:0/005> vbmeta_samsung.img
<ID:0/005> metadata.img
<ID:0/005> misc.bin
<ID:0/005> boot.img
<ID:0/005> modem.bin
<ID:0/005> modem_debug.bin
<ID:0/005> cache.img
<ID:0/005> omr.img
<ID:0/005> prism.img
<ID:0/005> optics.img
<ID:0/005> RQT_CLOSE !!
<ID:0/005> RES OK !!
<ID:0/005> Removed!!
<ID:0/005> Remain Port ....  0 
<OSM> All threads completed. (succeed 1 / failed 0)
```

Magisk install log:

```
- Device platform: arm64-v8a
- Magisk Manager: f5593e05 (4835)
- Install target: 21.4 (21400)
- Processing tar file
-- Extracting: boot.img
-- Copying: recovery.img.lz4
-- Copying: dtbo.img.lz4
-- Copying: super.img.lz4
-- Copying: userdata.img.lz4
-- Patching: vbmeta.img
-- Writing: vbmeta.img
-- Copying: vbmeta_samsung.img.lz4
-- Copying: metadata.img.lz4
-- Copying: misc.bin.lz4
-- Copying: meta-data/
-- Copying: meta-data/fota.zip
2026+0 records in
2026+0 records out
2074624 bytes (1.9 M) copied, 0.041887 s, 47 M/s
1381+1 records in
1381+1 records out
1415016 bytes (1.3 M) copied, 0.021825 s, 62 M/s
- Unpacking boot image
Parsing boot image: [/data/user_de/0/com.topjohnwu.magisk/install/boot.img]
HEADER_VER      [2]
KERNEL_SZ       [32219168]
RAMDISK_SZ      [902337]
SECOND_SZ       [0]
RECOV_DTBO_SZ   [0]
DTB_SZ          [190040]
OS_VERSION      [10.0.0]
OS_PATCH_LEVEL  [2020-11]
PAGESIZE        [2048]
NAME            [SRPTE13A001]
CMDLINE         [androidboot.selinux=enforce]
CHECKSUM        [f6c0ebd3ca253c6aea5cbf42ce941c2e17a0cee8000000000000000000000000]
KERNEL_FMT      [raw]
RAMDISK_FMT     [gzip]
- Checking ramdisk status
Loading cpio: [ramdisk.cpio]
- Stock boot image detected
- Patching ramdisk
Loading cpio: [ramdisk.cpio]
Add entry [init] (0750)
Patch with flag KEEPVERITY=[true] KEEPFORCEENCRYPT=[true]
Loading cpio: [ramdisk.cpio.orig]
Backup mismatch entry: [init] -> [.backup/init]
Create directory [.backup] (0000)
Add entry [.backup/.magisk] (0000)
Dump cpio: [ramdisk.cpio]
Loading dtbs from [dtb]
Patch @ 001A9DC8 [821B8012]->[E2FF8F12]
Patch @ 0193D993 [736B69705F696E697472616D667300]->[77616E745F696E697472616D667300]
- Repacking boot image
Parsing boot image: [/data/user_de/0/com.topjohnwu.magisk/install/boot.img]
HEADER_VER      [2]
KERNEL_SZ       [32219168]
RAMDISK_SZ      [902337]
SECOND_SZ       [0]
RECOV_DTBO_SZ   [0]
DTB_SZ          [190040]
OS_VERSION      [10.0.0]
OS_PATCH_LEVEL  [2020-11]
PAGESIZE        [2048]
NAME            [SRPTE13A001]
CMDLINE         [androidboot.selinux=enforce]
CHECKSUM        [f6c0ebd3ca253c6aea5cbf42ce941c2e17a0cee8000000000000000000000000]
KERNEL_FMT      [raw]
RAMDISK_FMT     [gzip]
Repack to boot image: [new-boot.img]
HEADER_VER      [2]
KERNEL_SZ       [32219168]
RAMDISK_SZ      [1194314]
SECOND_SZ       [0]
RECOV_DTBO_SZ   [0]
DTB_SZ          [190040]
OS_VERSION      [10.0.0]
OS_PATCH_LEVEL  [2020-11]
PAGESIZE        [2048]
NAME            [SRPTE13A001]
CMDLINE         [androidboot.selinux=enforce]
CHECKSUM        [c65de2a0f01cd9216cb3886f2299cd60c5682f98000000000000000000000000]
-- Writing: boot.img

****************************
 Output file is written to
 /storage/emulated/0/Download/magisk_patched_dYTnV.tar
****************************
- All done!
```

**Comment by msmuenchen** [comment]
Here's the vbmeta.img (original and patched) as well as the vbmeta_samsung.img (original, did not get patched).

[Archiv 3.zip](https://github.com/topjohnwu/Magisk/files/5849162/Archiv.3.zip)

The boot.img files unfortunately are too big for Github to upload, even compressed they rack up 13MB.

**Comment by msmuenchen** [comment]
With the current Canary, the flash step fails at boot.img: "Only official released binaries are allowed to be flashed(VBMETA)".

**Comment by msmuenchen** [comment]
To copy from what I've written on XDA:

-KG State is "Prenormal" (probably "Knox Guard"), FRP LOCK is OFF, OEM LOCK is OFF (U), Secure Download is Enabled (whatever that means)
- Replacing the vbmeta.img and vbmeta_samsung.img with an empty one (created with avbtool info_image --image empty.img) doesn't help, still the same error about "only official released binaries allowed to be flashed
- A really nasty problem: removing super.img and userdata.img to reduce the file copy and flash times from the stock AP image and to prevent ODIN from wiping out all user data (and thus, also from the Magisk patched image) leads to a corrupt system. Here's what I did: a flash of the full stock AP image, followed by booting into Android, doing the first-time initialization, rebooting to ODIN mode, and then flashing stock AP image sans userdata.img / super.img yields a prompt_and_wipe_data reason=set_policy_failed:/data/misc error in last_history. So, then I wiped the data in recovery, completed the install assistant - suddenly, it crashes and goes to recovery again, this time with a postrecovery_failed / rason=dataresizing_failed / RecoverySystemdataresizing_failed in /cache/recovery/last_history.
- Removing only userdata.img seems to work, but of course it takes *ages* to copy files between my Windows machine where Odin runs and my Mac

**Comment by msmuenchen** [comment]
I ran yet another attempt, using canary b76c80e2 and Samsung's latest firmware T575XXU2AUA4.

I am again getting this error:
```
ODIN MODE (AVB Fail)
vbmeta: Error verifying vbmeta image: HASH_MISMATCH (3)
vbmeta: VERIFICATION_DISABLED bit is set
CUSTOM VBMETA
VBMETA T575XXU2AUA4, 35608539R
```

The whole image tar file (sans super.img for size reasons, it's untouched anyway, and userdata.img) is at my Dropbox: https://www.dropbox.com/s/gymi91r8y2r9rzo/T575XXU2AUA4.zip?dl=0

Here's some more info (Magisk install log, mount output):

```
gtactive3:/ $ cat /sdcard/Download/magisk_install_log_2021-02-16T22\:07\:47Z.log
- Device platform: arm64-v8a
- Installing: b76c80e2 (21410)
- Processing tar file
-- Extracting: boot.img
-- Copying: dtbo.img.lz4
-- Copying: metadata.img.lz4
-- Copying: misc.bin.lz4
-- Copying: recovery.img.lz4
-- Copying: super.img.lz4
-- Patching: vbmeta.img
-- Writing: vbmeta.img
-- Copying: vbmeta_samsung.img.lz4
-- Copying: meta-data/
-- Copying: meta-data/fota.zip
- Unpacking boot image
Parsing boot image: [/data/user_de/0/com.topjohnwu.magisk/install/boot.img]
HEADER_VER      [2]
KERNEL_SZ       [32219168]
RAMDISK_SZ      [902327]
SECOND_SZ       [0]
RECOV_DTBO_SZ   [0]
DTB_SZ          [190040]
OS_VERSION      [10.0.0]
OS_PATCH_LEVEL  [2021-02]
PAGESIZE        [2048]
NAME            [SRPTE13A002]
CMDLINE         [androidboot.selinux=enforce]
CHECKSUM        [bdafd194c9b73f7f47a992ccd0d5092bd29a58a1000000000000000000000000]
KERNEL_FMT      [raw]
RAMDISK_FMT     [gzip]
- Checking ramdisk status
Loading cpio: [ramdisk.cpio]
- Stock boot image detected
- Patching ramdisk
Loading cpio: [ramdisk.cpio]
Add entry [init] (0750)
Create directory [overlay.d] (0750)
Create directory [overlay.d/sbin] (0750)
Add entry [overlay.d/sbin/magisk32.xz] (0644)
Add entry [overlay.d/sbin/magisk64.xz] (0644)
Patch with flag KEEPVERITY=[true] KEEPFORCEENCRYPT=[true]
Loading cpio: [ramdisk.cpio.orig]
Backup mismatch entry: [init] -> [.backup/init]
Record new entry: [overlay.d] -> [.backup/.rmlist]
Record new entry: [overlay.d/sbin] -> [.backup/.rmlist]
Record new entry: [overlay.d/sbin/magisk32.xz] -> [.backup/.rmlist]
Record new entry: [overlay.d/sbin/magisk64.xz] -> [.backup/.rmlist]
Create directory [.backup] (0000)
Add entry [.backup/.magisk] (0000)
Dump cpio: [ramdisk.cpio]
Loading dtbs from [dtb]
Patch @ 001A9DC8 [821B8012]->[E2FF8F12]
Patch @ 0193D993 [736B69705F696E697472616D667300]->[77616E745F696E697472616D667300]
Parsing boot image: [/data/user_de/0/com.topjohnwu.magisk/install/boot.img]
HEADER_VER      [2]
KERNEL_SZ       [32219168]
RAMDISK_SZ      [902327]
SECOND_SZ       [0]
RECOV_DTBO_SZ   [0]
DTB_SZ          [190040]
OS_VERSION      [10.0.0]
OS_PATCH_LEVEL  [2021-02]
PAGESIZE        [2048]
NAME            [SRPTE13A002]
CMDLINE         [androidboot.selinux=enforce]
CHECKSUM        [bdafd194c9b73f7f47a992ccd0d5092bd29a58a1000000000000000000000000]
- Repacking boot image
KERNEL_FMT      [raw]
RAMDISK_FMT     [gzip]
Repack to boot image: [new-boot.img]
HEADER_VER      [2]
KERNEL_SZ       [32219168]
RAMDISK_SZ      [1390118]
SECOND_SZ       [0]
RECOV_DTBO_SZ   [0]
DTB_SZ          [190040]
OS_VERSION      [10.0.0]
OS_PATCH_LEVEL  [2021-02]
PAGESIZE        [2048]
NAME            [SRPTE13A002]
CMDLINE         [androidboot.selinux=enforce]
CHECKSUM        [f3c00a5524109cda065507be9f8b11a4e168af2c000000000000000000000000]
-- Writing: boot.img

****************************
 Output file is written to
 /storage/emulated/0/Download/magisk_patched_goKgd.tar
****************************
cp: can't preserve ownership of 'busybox': Operation not permitted
cp: can't preserve ownership of 'magisk32': Operation not permitted
cp: can't preserve ownership of 'magisk64': Operation not permitted
cp: can't preserve ownership of 'magiskboot': Operation not permitted
cp: can't preserve ownership of 'magiskinit': Operation not permitted
- All done!
gtactive3:/ $ cat /proc/mounts
/dev/block/dm-4 / ext4 ro,seclabel,relatime,i_version 0 0
tmpfs /dev tmpfs rw,seclabel,nosuid,relatime,size=1829552k,nr_inodes=457388,mode=755 0 0
devpts /dev/pts devpts rw,seclabel,relatime,mode=600,ptmxmode=000 0 0
proc /proc proc rw,relatime,gid=3009,hidepid=2 0 0
sysfs /sys sysfs rw,seclabel,relatime 0 0
selinuxfs /sys/fs/selinux selinuxfs rw,relatime 0 0
tmpfs /mnt tmpfs rw,seclabel,nosuid,nodev,noexec,relatime,size=1829552k,nr_inodes=457388,mode=755,gid=1000 0 0
tmpfs /apex tmpfs rw,seclabel,nosuid,nodev,noexec,relatime,size=1829552k,nr_inodes=457388,mode=755 0 0
/dev/block/platform/11120000.ufs/by-name/metadata /metadata ext4 rw,seclabel,nosuid,nodev,noatime,journal_checksum,noauto_da_alloc,i_version 0 0
/dev/block/dm-5 /vendor ext4 ro,seclabel,relatime,i_version 0 0
/dev/block/dm-6 /product ext4 ro,seclabel,relatime,i_version 0 0
/dev/block/dm-7 /odm ext4 ro,seclabel,relatime,i_version 0 0
/dev/block/dm-8 /prism ext4 ro,seclabel,relatime,i_version 0 0
/dev/block/dm-9 /optics ext4 ro,seclabel,relatime,i_version 0 0
none /dev/cg2_bpf cgroup2 rw,nosuid,nodev,noexec,relatime 0 0
none /dev/cpuctl cgroup rw,nosuid,nodev,noexec,relatime,cpu 0 0
none /acct cgroup rw,nosuid,nodev,noexec,relatime,cpuacct 0 0
none /dev/cpuset cgroup rw,nosuid,nodev,noexec,relatime,cpuset,noprefix,release_agent=/sbin/cpuset_release_agent 0 0
none /dev/memcg cgroup rw,nosuid,nodev,noexec,relatime,memory 0 0
none /dev/stune cgroup rw,nosuid,nodev,noexec,relatime,schedtune 0 0
/sys/kernel/debug /sys/kernel/debug debugfs rw,seclabel,relatime 0 0
none /config configfs rw,nosuid,nodev,noexec,relatime 0 0
tmpfs /mnt/secure/asec tmpfs rw,seclabel,relatime,size=1833488k,nr_inodes=458372,mode=700 0 0
tmpfs /mnt/sde tmpfs rw,seclabel,relatime,size=1833488k,nr_inodes=458372,mode=700 0 0
bpf /sys/fs/bpf bpf rw,nosuid,nodev,noexec,relatime 0 0
pstore /sys/fs/pstore pstore rw,seclabel,nosuid,nodev,noexec,relatime 0 0
none /sys/kernel/config configfs rw,relatime 0 0
adb /dev/usb-ffs/adb functionfs rw,relatime 0 0
tracefs /sys/kernel/debug/tracing tracefs rw,seclabel,relatime 0 0
/dev/block/by-name/omr /omr ext4 rw,seclabel,nosuid,nodev,noatime,journal_checksum,noauto_da_alloc,i_version 0 0
/dev/block/platform/11120000.ufs/by-name/cpefs /mnt/vendor/cpefs ext4 rw,seclabel,nosuid,nodev,noatime,journal_checksum,noauto_da_alloc,i_version 0 0
/dev/block/platform/11120000.ufs/by-name/cache /cache ext4 rw,seclabel,nosuid,nodev,noatime,journal_checksum,noauto_da_alloc,i_version 0 0
/dev/block/platform/11120000.ufs/by-name/efs /mnt/vendor/efs ext4 rw,seclabel,nosuid,nodev,noatime,journal_checksum,noauto_da_alloc,i_version 0 0
/dev/block/platform/11120000.ufs/by-name/keydata /keydata ext4 rw,seclabel,nosuid,nodev,noatime,journal_checksum,noauto_da_alloc,i_version 0 0
/dev/block/platform/11120000.ufs/by-name/keyrefuge /keyrefuge ext4 rw,seclabel,nosuid,nodev,noatime,journal_checksum,noauto_da_alloc,i_version 0 0
/dev/block/platform/11120000.ufs/by-name/sec_efs /efs ext4 rw,seclabel,nosuid,nodev,noatime,journal_checksum,noauto_da_alloc,i_version 0 0
tmpfs /storage tmpfs rw,seclabel,nosuid,nodev,noexec,relatime,size=1829552k,nr_inodes=457388,mode=755,gid=1000 0 0
/dev/block/platform/11120000.ufs/by-name/userdata /data f2fs rw,lazytime,seclabel,nosuid,nodev,noatime,background_gc=on,discard,no_heap,user_xattr,inline_xattr,acl,inline_data,inline_dentry,extent_cache,mode=adaptive,active_logs=6,reserve_root=32768,reserve_core=1280,resuid=0,resgid=5678,usrquota,grpquota,alloc_mode=default,fsync_mode=nobarrier 0 0
/dev/block/loop2 /apex/com.android.resolv@290000000 ext4 ro,dirsync,seclabel,nodev,noatime,i_version 0 0
/dev/block/loop2 /apex/com.android.resolv ext4 ro,dirsync,seclabel,nodev,noatime,i_version 0 0
/dev/block/loop3 /apex/com.android.apex.cts.shim@1 ext4 ro,dirsync,seclabel,nodev,noatime,i_version 0 0
/dev/block/loop3 /apex/com.android.apex.cts.shim ext4 ro,dirsync,seclabel,nodev,noatime,i_version 0 0
/dev/block/loop4 /apex/com.android.runtime@1 ext4 ro,dirsync,seclabel,nodev,noatime,i_version 0 0
/dev/block/loop4 /apex/com.android.runtime ext4 ro,dirsync,seclabel,nodev,noatime,i_version 0 0
/dev/block/loop5 /apex/com.android.media@301400200 ext4 ro,dirsync,seclabel,nodev,noatime,i_version 0 0
/dev/block/loop5 /apex/com.android.media ext4 ro,dirsync,seclabel,nodev,noatime,i_version 0 0
/dev/block/loop6 /apex/com.android.media.swcodec@301400900 ext4 ro,dirsync,seclabel,nodev,noatime,i_version 0 0
/dev/block/loop6 /apex/com.android.media.swcodec ext4 ro,dirsync,seclabel,nodev,noatime,i_version 0 0
/dev/block/loop7 /apex/com.android.tzdata@293100000 ext4 ro,dirsync,seclabel,nodev,noatime,i_version 0 0
/dev/block/loop7 /apex/com.android.tzdata ext4 ro,dirsync,seclabel,nodev,noatime,i_version 0 0
/dev/block/loop8 /apex/com.android.conscrypt@290000000 ext4 ro,dirsync,seclabel,nodev,noatime,i_version 0 0
/dev/block/loop8 /apex/com.android.conscrypt ext4 ro,dirsync,seclabel,nodev,noatime,i_version 0 0
none /dev/freezer cgroup rw,relatime,freezer 0 0
/data/knox/secure_fs/enc_media /mnt/shell/enc_emulated sdcardfs rw,nosuid,nodev,noexec,noatime,fsuid=1000,fsgid=1000,gid=9997,multiuser,derive_gid,default_normal,reserved=20MB 0 0
/data/media /mnt/runtime/default/emulated sdcardfs rw,nosuid,nodev,noexec,noatime,fsuid=1023,fsgid=1023,gid=1015,multiuser,mask=6,derive_gid,default_normal,reserved=20MB,unshared_obb 0 0
/data/media /storage/emulated sdcardfs rw,nosuid,nodev,noexec,noatime,fsuid=1023,fsgid=1023,gid=1015,multiuser,mask=6,derive_gid,default_normal,reserved=20MB,unshared_obb 0 0
/data/media /mnt/runtime/read/emulated sdcardfs rw,nosuid,nodev,noexec,noatime,fsuid=1023,fsgid=1023,gid=9997,multiuser,mask=23,derive_gid,default_normal,reserved=20MB,unshared_obb 0 0
/data/media /mnt/runtime/write/emulated sdcardfs rw,nosuid,nodev,noexec,noatime,fsuid=1023,fsgid=1023,gid=9997,multiuser,mask=7,derive_gid,default_normal,reserved=20MB,unshared_obb 0 0
/data/media /mnt/runtime/full/emulated sdcardfs rw,nosuid,nodev,noexec,noatime,fsuid=1023,fsgid=1023,gid=9997,multiuser,mask=7,derive_gid,default_normal,reserved=20MB,unshared_obb 0 0
```

**Comment by msmuenchen** [comment]
To add: again, after reflashing the stock image I get straight booted into recovery:

```
Android Recovery
samsung/gtactive3xx/gtactive3
10/QP1A.190711.020/T575XXU2AUA4
user/release-keys
Android Recovery
samsung/gtactive3xx/gtactive3
10/QP1A.190711.020/T575XXU2AUA4
user/release-keys
Can't load Android system. Your data may be corrupt. If you continue to get this message, you may need to perform a factory data reset and erase all user data stored on this device.
[menu]
Reboot Recovery Cause is [UNKNOWN]
Reason is [set_policy_failed:/data/misc]
```

No way to get around that one.

**Comment by msmuenchen** [comment]
Ok, yet another trial... this time flashing everything, the whole package, with CSC instead of HOME_CSC... and another error, this time "Only official released binaries are allowed to be flashed (VBMETA)".

**Comment by jjgh** [comment]
@msmuenchen I think I'm having the same issue! Please see https://github.com/topjohnwu/Magisk/issues/3947
What can we do now?

**Comment by msmuenchen** [comment]
@jjgh take the Magisk modified tar file. Replace vbmeta.img with a zero-byte file, pack it together again, flash it. This is enough for rooting the SM-T575.

But, before that (!!!) check if KnoxGuard got tripped... power down the phone, connect it to USB, then press volup+voldn together. It should not show "KG STATE" at all. If it does, especially if KG STATE is "Checking" or "Prenormal", flash stock with CSC (basically, wipe it), *connect it to the internet during setup* and keep it continously online and powered for at least 7 days / 168 hours. Only then you will be able to flash.

@topjohnwu can you please create some sort of quirk for this tablet's vbmeta.img?

**Comment by osm0sis** [comment]
Looks like a duplicate of https://github.com/topjohnwu/Magisk/issues/3482 but this one has more useful troubleshooting info, so I'll close that one. 😉

**Comment by jjgh** [comment]
> check if KnoxGuard got tripped

@msmuenchen thanks for your support! With regards to the above sentence, did you mean I should make sure that KnoxGuard *is* tripped or that it is *not* tripped? (sorry for the dumb question but I am just trying to understand)

> It should not show "KG STATE" at all

Ok but what should it say instead of "KG STATE"? What should I see? 
I mean, how do I know the device is in a "good state" so that I can go on flashing the modified tar file?

Thank you!

**Comment by osm0sis** [comment]
Any change with latest Canary? Magisk now patches AVB flags in the images to disable vbmeta verification.

**Comment by jjgh** [comment]
@osm0sis how do I get the Canary build? 
Also, if this does not bother you too much, would you mind please trying to answer the two questions I asked @msmuenchen in the previous comment above? Perhaps the described procedure is really straightforward but I'd like to make sure before messing things up ;) thank you!

**Comment by osm0sis** [comment]
Same way as always, install Canary Magisk app and then install.

**Comment by jjgh** [comment]
@osm0sis I see but I can't find Canary version in this repo Releases? Or should I look somewhere else?
I apologize but since the vbmeta verification problem I haven't been using Magisk anymore so I kinda forgot the details...

**Comment by jjgh** [comment]
> You just go to the Downloads section of the readme... 

@osm0sis thanks

> 🤦‍♂️

but this was not very kind of you.

**Comment by L0neW0nderer** [comment]
@osm0sis I have a SM-T577 and ended up stuck at flashing Super.img. I had to go back to stock and now I have prenormal again which should take another week to resolve. I tried the steps in two phases, I used Canary and loaded that file only to get the hash message. I then went back and replaced vbmeta.img with a zero byte file and that's when things started getting stuck. 

Prior to this, I used a patched Odin along with two firmware with different regions. All ended with the same vbmeta hash message. I have to wonder if I made a mistake somewhere so I have to ask, how exactly was vbmeta supposed to be replaced with a zero byte file? Should the magisk patched ap file have a .md5 checksum?

**Comment by osm0sis** [comment]
That's a @topjohnwu question.

**Comment by teriblus** [comment]
@msmuenchen It seems you have been able to root the SM-T575.  I tried to follow your instructions but I face an issue.

1. I replaced vbmeta.img with a 0 byte file (touch vbmeta.img) 
2. I put it in the magisk patched AP tar (drag drop in archive with 7zip). 
3. Then I restarted the device in download mode (adb reboot download)
4. On download screen, I flashed with odin (patched AP with 0byte vbmeta + raw BL/CP/HOME)
5. I get an error saying that boot.img is not valid

Is there something I'm doing wrong ?
I used T575XXU3BUF1_T575OXM3BUF1_XEF firmware from sammobile.

**Comment by fredmobilinn** [comment]
Hello,
I did the same operations I was doing with succes on the Samsung Tab Active 2 (T395) using the last version of Odin  v3.14. and the lastest Canary version of Magisk and it failed with Samsung Tab Active 3 (T575). I get a message after Odin as done 90% of the copy process : "Installation blocked by OEM Lock" even if check the OEM unlock in the developper. The patched android image I used was T575XXU3BUL2. I used the the full original AP tar to patch it with Magisk

Any help ?
Thank you,

**Comment by bughunter2** [comment]
**Solution:**

For those stumbling upon this... I've had this with two different Samsung phones now. Both times, I was able to fix it.

I've posted the solution here:

https://github.com/topjohnwu/Magisk/issues/4395#issuecomment-2864564582
